dispute_gateway.go

```diff
@@ -0,0 +1,137 @@
+)
+
+type DisputeGateway struct {
+	*Braintree
+}
+
+type FileEvidenceRequest struct {
+	XMLName    xml.Name `xml:"evidence"`
+	DocumentId string   `xml:"document-id"`
+	Category   string   `xml:"category,omitempty"`
+}
+
+type TextEvidenceRequest struct {
+	XMLName  xml.Name `xml:"evidence"`
+	Content  string   `xml:"comments"`
+	Category string   `xml:"category,omitempty"`
+}
+
+func (g *DisputeGateway) AddFileEvidence(ctx context.Context, disputeId string, fileEvidenceRequest *FileEvidenceRequest) (*DisputeEvidence, error) {
+	resp, err := g.executeVersion(ctx, "POST", "disputes/"+disputeId+"/evidence", fileEvidenceRequest, apiVersion4)
+	if err != nil {
+		return nil, err
+	}
+	switch resp.StatusCode {
+	case 200:
+		return resp.disputeEvidence()
+	}
+	return nil, &invalidResponseError{resp}
+}
+
+func (g *DisputeGateway) AddTextEvidence(ctx context.Context, disputeId string, textEvidenceRequest *TextEvidenceRequest) (*DisputeEvidence, error) {
+	resp, err := g.executeVersion(ctx, "POST", "disputes/"+disputeId+"/evidence", textEvidenceRequest, apiVersion4)
+	if err != nil {
+		return nil, err
+	}
+	switch resp.StatusCode {
+	case 200:
+		return resp.disputeEvidence()
+	}
+	return nil, &invalidResponseError{resp}
+}
+
+func (g *DisputeGateway) Find(ctx context.Context, disputeId string) (*Dispute, error) {
+	resp, err := g.executeVersion(ctx, "GET", "disputes/"+disputeId, nil, apiVersion4)
+	if err != nil {
+		return nil, err
+	}
+	switch resp.StatusCode {
+	case 200:
+		return resp.dispute()
+	}
+	return nil, &invalidResponseError{resp}
+}
+
+func (g *DisputeGateway) Accept(ctx context.Context, disputeId string) (*Dispute, error) {
+	resp, err := g.executeVersion(ctx, "PUT", "disputes/"+disputeId+"/accept", nil, apiVersion4)
+	if err != nil {
+		return nil, err
+	}
+	switch resp.StatusCode {
+	case 200:
+		return resp.dispute()
+	}
+	return nil, &invalidResponseError{resp}
+}
+
+func (g *DisputeGateway) Finalize(ctx context.Context, disputeId string) (*Dispute, error) {
+	resp, err := g.executeVersion(ctx, "PUT", "disputes/"+disputeId+"/finalize", nil, apiVersion4)
+	if err != nil {
+		return nil, err
+	}
+	switch resp.StatusCode {
+	case 200:
+		return resp.dispute()
+	}
+	return nil, &invalidResponseError{resp}
+}
+
+func (g *DisputeGateway) RemoveEvidence(ctx context.Context, disputeId string, evidenceId string) error {
+	resp, err := g.executeVersion(ctx, "DELETE", "disputes/"+disputeId+"/evidence/"+evidenceId, nil, apiVersion4)
+	if err != nil {
+		return err
+	}
+	switch resp.StatusCode {
+	case 200:
+		return nil
+	}
+	return &invalidResponseError{resp}
+}
+
+func (g *DisputeGateway) Search(ctx context.Context, query *SearchQuery) (*DisputeSearchResult, error) {
+	resp, err := g.executeVersion(ctx, "POST", "dispute/advanced_search", query, apiVersion4)
+	if err != nil {
+		return nil, err
+	}
+	var v DisputeSearchResult
+	err = xml.Unmarshal(resp.Body, &v)
+	if err != nil {
+		return nil, err
+	}
+	return &v, err
+}
```

Comment:
I think we'll need a couple more `Search` functions. `SearchIDs`, `Search` and `SearchNext`, like here:

https://github.com/lionelbarrow/braintree-go/blob/273799669d13c0589383b83d45239c5b5819528a/transaction_gateway.go#L156-L158

https://github.com/lionelbarrow/braintree-go/blob/273799669d13c0589383b83d45239c5b5819528a/transaction_gateway.go#L181-L183

https://github.com/lionelbarrow/braintree-go/blob/273799669d13c0589383b83d45239c5b5819528a/transaction_gateway.go#L212-L215

But I think that's a pretty large piece of work and I think it's worth dropping the `Search` function for now, and lets do that in a separate PR once this is merged. The main goal for doing some of these other functions now is to make sure the modeling of `Dispute`, `DisputeEvidence`, `DisputeTransactionDetails`, and we should get that from the other functions you're adding in the gateway.

Reply:
@leighmcculloch did not find such implementations even in official Java client. Do such API handles exist (can't find the way to verify)?